lightning/src/ln/channelmanager.rs

```diff
@@ -8905,7 +8923,7 @@ macro_rules! create_refund_builder { ($self: ident, $builder: ty) => {
 	pub fn create_refund_builder(
 		&$self, amount_msats: u64, absolute_expiry: Duration, payment_id: PaymentId,
 		retry_strategy: Retry, max_total_routing_fee_msat: Option<u64>
-	) -> Result<$builder, Bolt12SemanticError> {
+	) -> Result<$builder, Bolt12CreationError> {
```

Comment:
Let's not change this to `Bolt12CreationError` if a few commits later it is changed to `Bolt12RequestError`. You can insert a fixup commit here to revert the change, which can be squashed before merging. Similarly for `pay_for_offer`.